src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -541,6 +541,184 @@ public String Insert(int startIndex, String value)
             }
             return result;
         }
+
+        public static string Join(char separator, params string[] value)
```

Comment:
I kept the parameter names the same as the string-based overloads, so this is `value` as opposed to `values`